### PR TITLE
Bitcoind and Electrum Server backend support

### PR DESCRIPTION
**Pull request by softsimon**
Work in progress.

**The following issues has been identified:**

- [x] Support InnerScripts (to display multisig etc) https://github.com/Blockstream/electrs/commit/5cd986896a7e7039a3f929ba9cd63ee1d05cef4f
- [x] Make a backend sync progress indicator
- [x] Segwit tag is not properly recognized 
- [x] Cannot load the Coinbase block transactions due to some limitation
- [x] Outspends are not displayed. Display or show as grey?

**NOT fixed and pushed to later versions**
- [ ] Lazy update block fees on first initial sync
- [ ] Address page balances not able to display sent and received, only balance

**How to**
Backend config:
```
MEMPOOL: {
    BACKEND: 'esplora' | 'electrum' | 'none';
},
"ESPLORA": {
    "REST_API_URL": "http://127.0.0.1:3000"
  },
  "ELECTRUM": {
    "HOST": "127.0.0.1",
    "PORT": 50002,
    "TLS_ENABLED": true,
    "TX_LOOKUPS": false
  },
  "BITCOIND": {
    "HOST": "127.0.0.1",
    "PORT": 3306,
    "USERNAME": "mempool",
    "PASSWORD": "mempool"
  },
```

Frontend config:
`  "ELECTRS_ITEMS_PER_PAGE": 10,`

### CODE REVIEW

**Comment by wiz** [comment]
Starting to test now - won't this break Liquid network support? Or should I simply add my elements core node RPC configuration under the "bitcoind" section? Since elements is a fork of bitcoin I imagine it mostly works but probably has random issues... in any case we shouldn't call that configuration section "bitcoind" but maybe something more generic

**Comment by wiz** [comment]
if (mempool.network === "liquid") default core_rpc port should be set to 7042

**Comment by wiz** [comment]
if (mempool.network === "testnet") default core_rpc port should be set to 18332

File: backend/package.json [hunk 1]
```diff
@@ -27,7 +27,12 @@
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "dependencies": {
+    "@codewarriorr/electrum-client-js": "^0.1.1",
```
Comment (wiz) [inline backend/package.json:30]:
```suggestion
```

**Review by wiz — APPROVED** [review]
ACK - tested OK on mempool.ninja with Mainnet, Testnet, Liquid, and Bisq all seems fine

**Comment by Kixunil** [comment]
Great feature! Would be nice to document this more visibly.

**Pull request merged**